# Webscraping

We will here try to do the webscrapping on the website https://www.365.be/fr/. This website have all the different activities present in Wallonia and in the region of Brussels capital. We will first import the different package needed for this webscrapping.

In [1]:
from IPython.display import display
import json
import pandas as pd
pd.options.display.max_columns = None
pd.options.display.max_rows = 700
from pprint import pprint
import re
import time
import os
import requests
from bs4 import BeautifulSoup
from selenium import webdriver

In [2]:
pip install selenium

Note: you may need to restart the kernel to use updated packages.


After that, we will collect the different informations given by the website. We will make a list that stores the information about attractions extracted from the website. Each code is for a different type of attractions. Each attraction's information is stored as a dictionary containing the name of the attraction, the type of attraction, the city where the attraction is and the price for each groups of persons. We will use selenium because we have here a dynamic website.

First Loop in order to have the different informations for Abbeys and religious monuments.

In [5]:
path = "C:Users/flori/Desktop/Master 1/Q2/Data Management"
driver = webdriver.Chrome(executable_path=os.path.join(path, 'chromedriver'))
attractions_infos1 = []
url = "https://www.365.be/fr/recherche/?category=1062&range=150&subCategory=1101"
driver.get(url)
time.sleep(5)

soup = BeautifulSoup(driver.page_source, 'html.parser')

events = soup.find('div', class_='main-content__body').find('div', class_='row cards').find_all('article', class_='col-12 col-md-6 col-lg-6 col-xl-4')

for event in events:
    full_event_url = f"https://www.365.be{event.find('a')['href']}"
    driver.get(full_event_url)
    time.sleep(2)
    soup = BeautifulSoup(driver.page_source, 'html.parser')
    
    try:
        attraction_name = soup.find('h1', class_="attraction-name").text.strip()
    except:
        attraction_name = "Not available"
    try:
        attraction_type = "Abbeys and religious monuments"
    except:
        attraction_type = "Not available"
    try:
        city_name = soup.find('span', {'itemprop': 'addressLocality'}).text.strip()
    except:
        city_name = "Not available"
    try:
        price_label = soup.find("span", class_="label", text="Prix").find_next_sibling("div").text.strip()
    except:
        price_label = "Not available"
    attractions_infos1.append({'Name' : attraction_name,'Type': attraction_type, 'City' : city_name, 'Price' : price_label})
print(attractions_infos1)
driver.quit()

C:\Users\flori\AppData\Local\Temp\ipykernel_16500\1129813387.py:2: DeprecationWarning: executable_path has been deprecated, please pass in a Service object
  driver = webdriver.Chrome(executable_path=os.path.join(path, 'chromedriver'))
Incompatible release of chromedriver (version 110.0.5481.77) detected in PATH: C:\Users\flori\Desktop\Master 1\Q2\Data Management\chromedriver.exe


[{'Name': 'Abbaye de Stavelot', 'Type': 'Abbeys and religious monuments', 'City': 'Stavelot', 'Price': 'Forfait All-in: adultes 11€ • 60+, enfants (6-18) 9,50€'}, {'Name': 'Abbaye de Villers', 'Type': 'Abbeys and religious monuments', 'City': 'Villers-la-Ville', 'Price': '(Plan de visite et accès aux jardins compris) : adultes 9€ - seniors, étudiants 7€ - enfants (6-12) 4€ - enfants (<6) gratuit'}, {'Name': 'Hôpital Notre-Dame à la Rose', 'Type': 'Abbeys and religious monuments', 'City': 'Lessines', 'Price': 'Ticket combiné pour la visite guidée ou audioguidée (musée et jardins): adultes 13€ • enfants (6-12 ans) 8€ • enfants (<6 ans)\xa0gratuit • Audioguides pour le musée disponibles en FR, NL, EN, DE, IT, ES • Audioguides pour les jardins disponibles seulement en FR / NL'}, {'Name': "Abbaye d'Orval", 'Type': 'Abbeys and religious monuments', 'City': 'Villers-devant-Orval', 'Price': 'adultes\xa07€ • 60+, étudiants 5€ • enfants (7-14) 3€'}, {'Name': 'Musée de la Vie wallonne', 'Type': '

Loop in order to have the different informations for activity related to archaeology.

In [7]:
path = "C:Users/flori/Desktop/Master 1/Q2/Data Management"
driver = webdriver.Chrome(executable_path=os.path.join(path, 'chromedriver'))
attractions_infos2 = []
url = "https://www.365.be/fr/recherche/?category=1062&range=150&subCategory=9942"
driver.get(url)
time.sleep(5)

soup = BeautifulSoup(driver.page_source, 'html.parser')

events = soup.find('div', class_='main-content__body').find('div', class_='row cards').find_all('article', class_='col-12 col-md-6 col-lg-6 col-xl-4')

for event in events:
    full_event_url = f"https://www.365.be{event.find('a')['href']}"
    driver.get(full_event_url)
    time.sleep(2)
    soup = BeautifulSoup(driver.page_source, 'html.parser')
    
    try:
        attraction_name = soup.find('h1', class_="attraction-name").text.strip()
    except:
        attraction_name = "Not available"
    try:
        attraction_type = "Archaeology"
    except:
        attraction_type = "Not available"
    try:
        city_name = soup.find('span', {'itemprop': 'addressLocality'}).text.strip()
    except:
        city_name = "Not available"
    try:
        price_label = soup.find("span", class_="label", text="Prix").find_next_sibling("div").text.strip()
    except:
        price_label = "Not available"
    attractions_infos2.append({'Name' : attraction_name,'Type': attraction_type, 'City' : city_name, 'Price' : price_label})
print(attractions_infos2)
driver.quit()

C:\Users\flori\AppData\Local\Temp\ipykernel_16500\1241346912.py:2: DeprecationWarning: executable_path has been deprecated, please pass in a Service object
  driver = webdriver.Chrome(executable_path=os.path.join(path, 'chromedriver'))
Incompatible release of chromedriver (version 110.0.5481.77) detected in PATH: C:\Users\flori\Desktop\Master 1\Q2\Data Management\chromedriver.exe


[{'Name': 'Préhistomuseum', 'Type': 'Archaeology', 'City': 'Flémalle', 'Price': 'PREHISTOMUSEUM : enfants 9€ • étudiants et 65+ 14€ • adultes 15€   Supplément pour les expositions temporaires   TRIBU (4 à 15 personnes) = Réduction -10%'}, {'Name': 'Espace gallo-romain', 'Type': 'Archaeology', 'City': 'Ath', 'Price': "adultes et seniors 6€ • étudiants (<26) sur présentation de la carte : 4€ • PMR, enfants et jeunes (3-18) 4€ • enfants (<3) gratuit • Entrée gratuite le 1er dimanche du mois (toute l'année)"}, {'Name': 'Musée du Malgré-Tout. Parc de la Préhistoire', 'Type': 'Archaeology', 'City': 'Treignes', 'Price': 'adultes 5€ • 60+, étudiants 4€ • enfants (6-12) 2,50€ • enfants (<6) gratuit'}, {'Name': 'Malagne - Archéoparc de Rochefort', 'Type': 'Archaeology', 'City': 'Rochefort', 'Price': '(hors événements) : adultes 7,50€ •\xa060+ 6,50€ • enfants 6€  \xa0Audioguide FR, NL, EN offert'}, {'Name': 'Musée du Château Fort de Logne', 'Type': 'Archaeology', 'City': 'Vieuxville', 'Price': '(

Loop in order to have the different informations for activity related to art.

In [9]:
path = "C:Users/flori/Desktop/Master 1/Q2/Data Management"
driver = webdriver.Chrome(executable_path=os.path.join(path, 'chromedriver'))
attractions_infos3 = []
url = "https://www.365.be/fr/recherche/?category=1062&range=150&subCategory=1070"
driver.get(url)
time.sleep(5)

soup = BeautifulSoup(driver.page_source, 'html.parser')

events = soup.find('div', class_='main-content__body').find('div', class_='row cards').find_all('article', class_='col-12 col-md-6 col-lg-6 col-xl-4')

for event in events:
    full_event_url = f"https://www.365.be{event.find('a')['href']}"
    driver.get(full_event_url)
    time.sleep(2)
    soup = BeautifulSoup(driver.page_source, 'html.parser')
    
    try:
        attraction_name = soup.find('h1', class_="attraction-name").text.strip()
    except:
        attraction_name = "Not available"
    try:
        attraction_type = "Art"
    except:
        attraction_type = "Not available"
    try:
        city_name = soup.find('span', {'itemprop': 'addressLocality'}).text.strip()
    except:
        city_name = "Not available"
    try:
        price_label = soup.find("span", class_="label", text="Prix").find_next_sibling("div").text.strip()
    except:
        price_label = "Not available"
    attractions_infos3.append({'Name' : attraction_name,'Type': attraction_type, 'City' : city_name, 'Price' : price_label})
print(attractions_infos3)
driver.quit()

C:\Users\flori\AppData\Local\Temp\ipykernel_16500\396412053.py:2: DeprecationWarning: executable_path has been deprecated, please pass in a Service object
  driver = webdriver.Chrome(executable_path=os.path.join(path, 'chromedriver'))
Incompatible release of chromedriver (version 110.0.5481.77) detected in PATH: C:\Users\flori\Desktop\Master 1\Q2\Data Management\chromedriver.exe


[{'Name': 'Musée Hergé', 'Type': 'Art', 'City': 'Louvain-la-Neuve', 'Price': 'adultes 12€ • 65+, étudiants (avec carte) 7€ • Enfants (7-14) 5€ • enfants (<7) gratuit'}, {'Name': 'Musée L', 'Type': 'Art', 'City': 'Louvain-la-Neuve', 'Price': 'adultes 8€ • 60+ 6€ • étudiants (13-25) et PMR : 3€ • Article 27 : 1,25€ • enfants (<12) et enseignants : gratuit • Entrée et visioguide gratuits pour toutes et tous chaque 1er dimanche du mois'}, {'Name': 'Domaine du Château de Seneffe', 'Type': 'Art', 'City': 'Seneffe', 'Price': 'adultes 6€ -\xa060+ et étudiants (12-18) 5€ - enfants (<12) et PMR : gratuit\xa0  Le Parc et les Jardins sont accessibles gratuitement (hors activités)   Chaque 1er dimanche du mois accès gratuit à la collection permanente "D\'un jour à l\'autre, le XVIIIe" (mais accès payant pour l\'exposition temporaire)'}, {'Name': 'La Boverie', 'Type': 'Art', 'City': 'Liège', 'Price': 'adultes 5€ • 60+ 3€ • <26 ans : gratuit • Collection permanente : entrée gratuite le 1er dimanche d

Loop in order to have the different informations for activity related to the handicrafts and know-how.

In [11]:
path = "C:Users/flori/Desktop/Master 1/Q2/Data Management"
driver = webdriver.Chrome(executable_path=os.path.join(path, 'chromedriver'))
attractions_infos4 = []
url = "https://www.365.be/fr/recherche/?category=1062&range=150&subCategory=5414"
driver.get(url)
time.sleep(5)

soup = BeautifulSoup(driver.page_source, 'html.parser')

events = soup.find('div', class_='main-content__body').find('div', class_='row cards').find_all('article', class_='col-12 col-md-6 col-lg-6 col-xl-4')

for event in events:
    full_event_url = f"https://www.365.be{event.find('a')['href']}"
    driver.get(full_event_url)
    time.sleep(2)
    soup = BeautifulSoup(driver.page_source, 'html.parser')
    
    try:
        attraction_name = soup.find('h1', class_="attraction-name").text.strip()
    except:
        attraction_name = "Not available"
    try:
        attraction_type = "Handicrafts and know-how"
    except:
        attraction_type = "Not available"
    try:
        city_name = soup.find('span', {'itemprop': 'addressLocality'}).text.strip()
    except:
        city_name = "Not available"
    try:
        price_label = soup.find("span", class_="label", text="Prix").find_next_sibling("div").text.strip()
    except:
        price_label = "Not available"
    attractions_infos4.append({'Name' : attraction_name,'Type': attraction_type, 'City' : city_name, 'Price' : price_label})
print(attractions_infos4)
driver.quit()

C:\Users\flori\AppData\Local\Temp\ipykernel_16500\2477011148.py:2: DeprecationWarning: executable_path has been deprecated, please pass in a Service object
  driver = webdriver.Chrome(executable_path=os.path.join(path, 'chromedriver'))
Incompatible release of chromedriver (version 110.0.5481.77) detected in PATH: C:\Users\flori\Desktop\Master 1\Q2\Data Management\chromedriver.exe


[{'Name': 'Musée du Marbre', 'Type': 'Handicrafts and know-how', 'City': 'Rance', 'Price': 'adultes\xa06€ • étudiants, 60+ : 5€ • enfants (6-12) 3€ • enfants (<6) gratuit   Gratuit le 1er dimanche du mois • Accessible aux PMR. Art 27'}, {'Name': 'Malmundarium', 'Type': 'Handicrafts and know-how', 'City': 'Malmedy', 'Price': 'adultes 6€ • 60+, étudiants 5€ • enfants (6-12) et enseignants 3€ • enfants (0-5) gratuit   Carte familiale journalière (2 parents + enfants) : 15€ • Abonnement annuel individuel : 15€ • Abonnement annuel familial : 25€ • Accessible aux PMR'}, {'Name': 'Malagne - Archéoparc de Rochefort', 'Type': 'Handicrafts and know-how', 'City': 'Rochefort', 'Price': '(hors événements) : adultes 7,50€ •\xa060+ 6,50€ • enfants 6€  \xa0Audioguide FR, NL, EN offert'}, {'Name': 'Centre Touristique de la Laine et de la Mode', 'Type': 'Handicrafts and know-how', 'City': 'Verviers', 'Price': 'adultes 6€\xa0• 60+, étudiants (19-26) : 5€ • étudiants (6-18) 4€ • enfants (3-5) 1,50€   enfa

Loop in order to have the different informations for activity related to the castles, citadels and belfries.

In [14]:
path = "C:Users/flori/Desktop/Master 1/Q2/Data Management"
driver = webdriver.Chrome(executable_path=os.path.join(path, 'chromedriver'))
attractions_infos5 = []
url = "https://www.365.be/fr/recherche/?category=1062&range=150&subCategory=9943"
driver.get(url)
time.sleep(5)

soup = BeautifulSoup(driver.page_source, 'html.parser')

events = soup.find('div', class_='main-content__body').find('div', class_='row cards').find_all('article', class_='col-12 col-md-6 col-lg-6 col-xl-4')

for event in events:
    full_event_url = f"https://www.365.be{event.find('a')['href']}"
    driver.get(full_event_url)
    time.sleep(2)
    soup = BeautifulSoup(driver.page_source, 'html.parser')
    
    try:
        attraction_name = soup.find('h1', class_="attraction-name").text.strip()
    except:
        attraction_name = "Not available"
    try:
        attraction_type = "Castles, citadels and belfries"
    except:
        attraction_type = "Not available"
    try:
        city_name = soup.find('span', {'itemprop': 'addressLocality'}).text.strip()
    except:
        city_name = "Not available"
    try:
        price_label = soup.find("span", class_="label", text="Prix").find_next_sibling("div").text.strip()
    except:
        price_label = "Not available"
    attractions_infos5.append({'Name' : attraction_name,'Type': attraction_type, 'City' : city_name, 'Price' : price_label})
print(attractions_infos5)
driver.quit()

C:\Users\flori\AppData\Local\Temp\ipykernel_16500\3329595399.py:2: DeprecationWarning: executable_path has been deprecated, please pass in a Service object
  driver = webdriver.Chrome(executable_path=os.path.join(path, 'chromedriver'))
Incompatible release of chromedriver (version 110.0.5481.77) detected in PATH: C:\Users\flori\Desktop\Master 1\Q2\Data Management\chromedriver.exe


[{'Name': 'Citadelle de Namur', 'Type': 'Castles, citadels and belfries', 'City': 'Namur', 'Price': 'Citadelle Pass (train touristique, souterrains, Centre du visiteur Terra Nova) : adultes\xa018€ - enfants (<18), étudiants : 16€ - enfants (<6) gratuit • Tarif pour une seule visite guidée au choix (train touristique ou souterrains) : de 6 à 12€ • Centre du visiteur Terra Nova (non guidé) : adultes 6€ - enfants (<18), étudiants : 5€ - enfants (<6) gratuit'}, {'Name': 'Château fort de Bouillon', 'Type': 'Castles, citadels and belfries', 'City': 'Bouillon', 'Price': 'Bouillon City Pass + (du 1/04 au 11/11) : château fort + spectacle de fauconnerie + musée Scriptura + spectacle nocturne "L\'Odyssée de Lumière" + Archéoscope + Musée Ducal + expositions temporaires : adultes et 60+ : 15€ - enfants (4 à 12 ans) et étudiants : 11€ - enfants (< 4 ans) : gratuit   Bouillon City Pass (hors saison) : adultes et 60+ : 11€ - enfants (4 à 12 ans) et étudiants : 8,50€ - enfants (< 4 ans) : gratuit   L

Loop in order to have the different informations for activity related to the Folklore and local history.

In [16]:
path = "C:Users/flori/Desktop/Master 1/Q2/Data Management"
driver = webdriver.Chrome(executable_path=os.path.join(path, 'chromedriver'))
attractions_infos6 = []
url = "https://www.365.be/fr/recherche/?category=1062&range=150&subCategory=9944"
driver.get(url)
time.sleep(5)

soup = BeautifulSoup(driver.page_source, 'html.parser')

events = soup.find('div', class_='main-content__body').find('div', class_='row cards').find_all('article', class_='col-12 col-md-6 col-lg-6 col-xl-4')

for event in events:
    full_event_url = f"https://www.365.be{event.find('a')['href']}"
    driver.get(full_event_url)
    time.sleep(2)
    soup = BeautifulSoup(driver.page_source, 'html.parser')
    
    try:
        attraction_name = soup.find('h1', class_="attraction-name").text.strip()
    except:
        attraction_name = "Not available"
    try:
        attraction_type = "Folklore and local history"
    except:
        attraction_type = "Not available"
    try:
        city_name = soup.find('span', {'itemprop': 'addressLocality'}).text.strip()
    except:
        city_name = "Not available"
    try:
        price_label = soup.find("span", class_="label", text="Prix").find_next_sibling("div").text.strip()
    except:
        price_label = "Not available"
    attractions_infos6.append({'Name' : attraction_name,'Type': attraction_type, 'City' : city_name, 'Price' : price_label})
print(attractions_infos6)
driver.quit()

C:\Users\flori\AppData\Local\Temp\ipykernel_16500\3700513510.py:2: DeprecationWarning: executable_path has been deprecated, please pass in a Service object
  driver = webdriver.Chrome(executable_path=os.path.join(path, 'chromedriver'))
Incompatible release of chromedriver (version 110.0.5481.77) detected in PATH: C:\Users\flori\Desktop\Master 1\Q2\Data Management\chromedriver.exe


[{'Name': 'Malmundarium', 'Type': 'Folklore and local history', 'City': 'Malmedy', 'Price': 'adultes 6€ • 60+, étudiants 5€ • enfants (6-12) et enseignants 3€ • enfants (0-5) gratuit   Carte familiale journalière (2 parents + enfants) : 15€ • Abonnement annuel individuel : 15€ • Abonnement annuel familial : 25€ • Accessible aux PMR'}, {'Name': 'Espace Arthur Masson', 'Type': 'Folklore and local history', 'City': 'Treignes', 'Price': 'Visite ou animation : adultes 7€ - 60+ 6,50€ -\xa0étudiants (>18) 6€ - enfants (6-18) 5€ - enfants (<6) gratuit'}, {'Name': 'Musée de la Vie wallonne', 'Type': 'Folklore and local history', 'City': 'Liège', 'Price': 'adultes : à partir de 7€ • 60+, étudiants, enfants (3-18) : à partir de 5€ • enfants (<6) gratuit   Entrée gratuite le 1er dimanche du mois'}, {'Name': 'Maison du patrimoine médiéval mosan', 'Type': 'Folklore and local history', 'City': 'Bouvignes (Dinant)', 'Price': '6€'}, {'Name': 'Maison des Géants', 'Type': 'Folklore and local history', 'C

Loop in order to have the different informations for activity related to the history.

In [18]:
path = "C:Users/flori/Desktop/Master 1/Q2/Data Management"
driver = webdriver.Chrome(executable_path=os.path.join(path, 'chromedriver'))
attractions_infos7 = []
url = "https://www.365.be/fr/recherche/?category=1062&range=150&subCategory=9945"
driver.get(url)
time.sleep(5)

soup = BeautifulSoup(driver.page_source, 'html.parser')

events = soup.find('div', class_='main-content__body').find('div', class_='row cards').find_all('article', class_='col-12 col-md-6 col-lg-6 col-xl-4')

for event in events:
    full_event_url = f"https://www.365.be{event.find('a')['href']}"
    driver.get(full_event_url)
    time.sleep(2)
    soup = BeautifulSoup(driver.page_source, 'html.parser')
    
    try:
        attraction_name = soup.find('h1', class_="attraction-name").text.strip()
    except:
        attraction_name = "Not available"
    try:
        attraction_type = "History"
    except:
        attraction_type = "Not available"
    try:
        city_name = soup.find('span', {'itemprop': 'addressLocality'}).text.strip()
    except:
        city_name = "Not available"
    try:
        price_label = soup.find("span", class_="label", text="Prix").find_next_sibling("div").text.strip()
    except:
        price_label = "Not available"
    attractions_infos7.append({'Name' : attraction_name,'Type': attraction_type, 'City' : city_name, 'Price' : price_label})
print(attractions_infos7)
driver.quit()

C:\Users\flori\AppData\Local\Temp\ipykernel_16500\1226437937.py:2: DeprecationWarning: executable_path has been deprecated, please pass in a Service object
  driver = webdriver.Chrome(executable_path=os.path.join(path, 'chromedriver'))
Incompatible release of chromedriver (version 110.0.5481.77) detected in PATH: C:\Users\flori\Desktop\Master 1\Q2\Data Management\chromedriver.exe


[{'Name': 'Hôpital Notre-Dame à la Rose', 'Type': 'History', 'City': 'Lessines', 'Price': 'Ticket combiné pour la visite guidée ou audioguidée (musée et jardins): adultes 13€ • enfants (6-12 ans) 8€ • enfants (<6 ans)\xa0gratuit • Audioguides pour le musée disponibles en FR, NL, EN, DE, IT, ES • Audioguides pour les jardins disponibles seulement en FR / NL'}, {'Name': 'Château fort de Bouillon', 'Type': 'History', 'City': 'Bouillon', 'Price': 'Bouillon City Pass + (du 1/04 au 11/11) : château fort + spectacle de fauconnerie + musée Scriptura + spectacle nocturne "L\'Odyssée de Lumière" + Archéoscope + Musée Ducal + expositions temporaires : adultes et 60+ : 15€ - enfants (4 à 12 ans) et étudiants : 11€ - enfants (< 4 ans) : gratuit   Bouillon City Pass (hors saison) : adultes et 60+ : 11€ - enfants (4 à 12 ans) et étudiants : 8,50€ - enfants (< 4 ans) : gratuit   Livret-guide : 1,50€   Audioguide : 2,50€'}, {'Name': 'Mundaneum', 'Type': 'History', 'City': 'Mons', 'Price': "adultes 9€ •

Loop in order to have the different informations for activity related to the museums and science parks.

In [20]:
path = "C:Users/flori/Desktop/Master 1/Q2/Data Management"
driver = webdriver.Chrome(executable_path=os.path.join(path, 'chromedriver'))
attractions_infos8 = []
url = "https://www.365.be/fr/recherche/?category=1062&range=150&subCategory=5413"
driver.get(url)
time.sleep(5)

soup = BeautifulSoup(driver.page_source, 'html.parser')

events = soup.find('div', class_='main-content__body').find('div', class_='row cards').find_all('article', class_='col-12 col-md-6 col-lg-6 col-xl-4')

for event in events:
    full_event_url = f"https://www.365.be{event.find('a')['href']}"
    driver.get(full_event_url)
    time.sleep(2)
    soup = BeautifulSoup(driver.page_source, 'html.parser')
    
    try:
        attraction_name = soup.find('h1', class_="attraction-name").text.strip()
    except:
        attraction_name = "Not available"
    try:
        attraction_type = "Museums and science parks"
    except:
        attraction_type = "Not available"
    try:
        city_name = soup.find('span', {'itemprop': 'addressLocality'}).text.strip()
    except:
        city_name = "Not available"
    try:
        price_label = soup.find("span", class_="label", text="Prix").find_next_sibling("div").text.strip()
    except:
        price_label = "Not available"
    attractions_infos8.append({'Name' : attraction_name,'Type': attraction_type, 'City' : city_name, 'Price' : price_label})
print(attractions_infos8)
driver.quit()

C:\Users\flori\AppData\Local\Temp\ipykernel_16500\3356107156.py:2: DeprecationWarning: executable_path has been deprecated, please pass in a Service object
  driver = webdriver.Chrome(executable_path=os.path.join(path, 'chromedriver'))
Incompatible release of chromedriver (version 110.0.5481.77) detected in PATH: C:\Users\flori\Desktop\Master 1\Q2\Data Management\chromedriver.exe


[{'Name': 'Euro Space Center', 'Type': 'Museums and science parks', 'City': 'Transinne', 'Price': 'Réservation en ligne souhaitée : adultes 33€ • 60+ 31€ • enfants (6-18 ans)\xa029€'}, {'Name': 'SPARKOH!', 'Type': 'Museums and science parks', 'City': 'Frameries', 'Price': 'adultes 20€ - Seniors (65+) 17€ - enfants (4-18) 15€ - enfants (<4) gratuit'}, {'Name': 'Maison de la Science', 'Type': 'Museums and science parks', 'City': 'Liège', 'Price': "Combiné avec l'Aquarium : adultes 13€ • 65+, PMR, étudiants (13-17) : 10€ • enfants (6-12) 8€ • enfants (<6) gratuit"}, {'Name': "Musée d'Histoire naturelle et Vivarium", 'Type': 'Museums and science parks', 'City': 'Tournai', 'Price': 'adultes 4€ • 65+, enfants, étudiants (6-18): 3€ • Gratuit le 1er dimanche du mois   Audioguide (FR, NL, EN) 2€'}, {'Name': 'Muséum des Sciences naturelles', 'Type': 'Museums and science parks', 'City': 'Bruxelles', 'Price': 'adultes 13€ • 65+, PMR 10€ • jeunes (4-17) et étudiants (18-23) : 5€ • enfants (<4) grat

Loop in order to have the different informations for activity related to the automotive and railway heritage.

In [23]:
path = "C:Users/flori/Desktop/Master 1/Q2/Data Management"
driver = webdriver.Chrome(executable_path=os.path.join(path, 'chromedriver'))
attractions_infos9 = []
url = "https://www.365.be/fr/recherche/?category=1062&range=150&subCategory=9946"
driver.get(url)
time.sleep(5)

soup = BeautifulSoup(driver.page_source, 'html.parser')

events = soup.find('div', class_='main-content__body').find('div', class_='row cards').find_all('article', class_='col-12 col-md-6 col-lg-6 col-xl-4')

for event in events:
    full_event_url = f"https://www.365.be{event.find('a')['href']}"
    driver.get(full_event_url)
    time.sleep(2)
    soup = BeautifulSoup(driver.page_source, 'html.parser')
    
    try:
        attraction_name = soup.find('h1', class_="attraction-name").text.strip()
    except:
        attraction_name = "Not available"
    try:
        attraction_type = "Automotive and railway heritage"
    except:
        attraction_type = "Not available"
    try:
        city_name = soup.find('span', {'itemprop': 'addressLocality'}).text.strip()
    except:
        city_name = "Not available"
    try:
        price_label = soup.find("span", class_="label", text="Prix").find_next_sibling("div").text.strip()
    except:
        price_label = "Not available"
    attractions_infos9.append({'Name' : attraction_name,'Type': attraction_type, 'City' : city_name, 'Price' : price_label})
print(attractions_infos9)
driver.quit()

C:\Users\flori\AppData\Local\Temp\ipykernel_16500\2385653867.py:2: DeprecationWarning: executable_path has been deprecated, please pass in a Service object
  driver = webdriver.Chrome(executable_path=os.path.join(path, 'chromedriver'))
Incompatible release of chromedriver (version 110.0.5481.77) detected in PATH: C:\Users\flori\Desktop\Master 1\Q2\Data Management\chromedriver.exe


[{'Name': 'Abbaye de Stavelot', 'Type': 'Automotive and railway heritage', 'City': 'Stavelot', 'Price': 'Forfait All-in: adultes 11€ • 60+, enfants (6-18) 9,50€'}, {'Name': 'Musée du chemin de fer à vapeur', 'Type': 'Automotive and railway heritage', 'City': 'Treignes', 'Price': 'adultes 6€ - 60+ 5€ - enfants (6-12)\xa04€ • Combiné Musée + Chemin de fer à vapeur des 3 vallées : adultes, 60+ 21€ - enfants (6-12) 14€'}, {'Name': 'Musée des Transports en Commun de Wallonie', 'Type': 'Automotive and railway heritage', 'City': 'Liège', 'Price': 'adultes\xa06€ • 60+, étudiants, PMR\xa05€ • enfants (6-12) 4€ • enfants (3-5) 1€ • enfants (<3) gratuit • Billet famille (2 adultes + max. 3 enfants) 16€'}, {'Name': 'Mahymobiles', 'Type': 'Automotive and railway heritage', 'City': 'Leuze-en-Hainaut', 'Price': 'adultes 15€ -\xa0enfants (<12) 10€ - enfants (<6) gratuit • Accessible aux PMR'}, {'Name': 'Autoworld', 'Type': 'Automotive and railway heritage', 'City': 'Bruxelles', 'Price': 'adultes 15€ •

Loop in order to have the different informations for activity related to the industrial heritage and engineering structures.

In [25]:
path = "C:Users/flori/Desktop/Master 1/Q2/Data Management"
driver = webdriver.Chrome(executable_path=os.path.join(path, 'chromedriver'))
attractions_infos10 = []
url = "https://www.365.be/fr/recherche/?category=1062&range=150&subCategory=9947"
driver.get(url)
time.sleep(5)

soup = BeautifulSoup(driver.page_source, 'html.parser')

events = soup.find('div', class_='main-content__body').find('div', class_='row cards').find_all('article', class_='col-12 col-md-6 col-lg-6 col-xl-4')

for event in events:
    full_event_url = f"https://www.365.be{event.find('a')['href']}"
    driver.get(full_event_url)
    time.sleep(2)
    soup = BeautifulSoup(driver.page_source, 'html.parser')
    
    try:
        attraction_name = soup.find('h1', class_="attraction-name").text.strip()
    except:
        attraction_name = "Not available"
    try:
        attraction_type = "Industrial heritage and engineering structures"
    except:
        attraction_type = "Not available"
    try:
        city_name = soup.find('span', {'itemprop': 'addressLocality'}).text.strip()
    except:
        city_name = "Not available"
    try:
        price_label = soup.find("span", class_="label", text="Prix").find_next_sibling("div").text.strip()
    except:
        price_label = "Not available"
    attractions_infos10.append({'Name' : attraction_name,'Type': attraction_type, 'City' : city_name, 'Price' : price_label})
print(attractions_infos10)
driver.quit()

C:\Users\flori\AppData\Local\Temp\ipykernel_16500\658102210.py:2: DeprecationWarning: executable_path has been deprecated, please pass in a Service object
  driver = webdriver.Chrome(executable_path=os.path.join(path, 'chromedriver'))
Incompatible release of chromedriver (version 110.0.5481.77) detected in PATH: C:\Users\flori\Desktop\Master 1\Q2\Data Management\chromedriver.exe


[{'Name': 'Musée du Marbre', 'Type': 'Industrial heritage and engineering structures', 'City': 'Rance', 'Price': 'adultes\xa06€ • étudiants, 60+ : 5€ • enfants (6-12) 3€ • enfants (<6) gratuit   Gratuit le 1er dimanche du mois • Accessible aux PMR. Art 27'}, {'Name': 'Barrage et Skywalk', 'Type': 'Industrial heritage and engineering structures', 'City': 'Boussu-lez-Walcourt', 'Price': 'adultes 12€ • enfants (4-12) 10€ • enfants (<4) gratuit'}, {'Name': 'Le Bois du Cazier', 'Type': 'Industrial heritage and engineering structures', 'City': 'Marcinelle', 'Price': 'adultes 9€\xa0•\xa060+ 8€ • Jeunes et étudiants : 5€\xa0•\xa0enfants (<6) : gratuit   Audioguide : 1€ • Livret-jeu : 1€ • Appli gratuite'}, {'Name': 'Ascenseur funiculaire de Strépy-Thieu et Croisière sur le Canal du Centre historique', 'Type': 'Industrial heritage and engineering structures', 'City': 'Thieu', 'Price': "Visite intérieure de l'Ascenseur funiculaire (centre d'interprétation et film) : adultes 10€ - 60+, étudiants 

Loop in order to have the different informations for activity related to the local products.

In [27]:
path = "C:Users/flori/Desktop/Master 1/Q2/Data Management"
driver = webdriver.Chrome(executable_path=os.path.join(path, 'chromedriver'))
attractions_infos11 = []
url = "https://www.365.be/fr/recherche/?category=1062&range=150&subCategory=9948"
driver.get(url)
time.sleep(5)

soup = BeautifulSoup(driver.page_source, 'html.parser')

events = soup.find('div', class_='main-content__body').find('div', class_='row cards').find_all('article', class_='col-12 col-md-6 col-lg-6 col-xl-4')

for event in events:
    full_event_url = f"https://www.365.be{event.find('a')['href']}"
    driver.get(full_event_url)
    time.sleep(2)
    soup = BeautifulSoup(driver.page_source, 'html.parser')
    
    try:
        attraction_name = soup.find('h1', class_="attraction-name").text.strip()
    except:
        attraction_name = "Not available"
    try:
        attraction_type = "Local products"
    except:
        attraction_type = "Not available"
    try:
        city_name = soup.find('span', {'itemprop': 'addressLocality'}).text.strip()
    except:
        city_name = "Not available"
    try:
        price_label = soup.find("span", class_="label", text="Prix").find_next_sibling("div").text.strip()
    except:
        price_label = "Not available"
    attractions_infos11.append({'Name' : attraction_name,'Type': attraction_type, 'City' : city_name, 'Price' : price_label})
print(attractions_infos11)
driver.quit()

C:\Users\flori\AppData\Local\Temp\ipykernel_16500\2180430144.py:2: DeprecationWarning: executable_path has been deprecated, please pass in a Service object
  driver = webdriver.Chrome(executable_path=os.path.join(path, 'chromedriver'))
Incompatible release of chromedriver (version 110.0.5481.77) detected in PATH: C:\Users\flori\Desktop\Master 1\Q2\Data Management\chromedriver.exe


[{'Name': 'Abbaye de Villers', 'Type': 'Local products', 'City': 'Villers-la-Ville', 'Price': '(Plan de visite et accès aux jardins compris) : adultes 9€ - seniors, étudiants 7€ - enfants (6-12) 4€ - enfants (<6) gratuit'}, {'Name': 'Chimay Experience', 'Type': 'Local products', 'City': 'Bourlers', 'Price': 'adultes 6,50€ (comprenant une Chimay au fût)\xa0• 60+, professeurs belges, étudiants (12-18) : 5€ • enfants (<12) gratuit • Accessible aux PMR'}, {'Name': "Musée de la Bière et du Peket. Château de l'Avouerie", 'Type': 'Local products', 'City': 'Anthisnes', 'Price': 'Visite et dégustation : adultes 7€\xa0• 65+, étudiants, enfants (3-15 ans) : 5€   Animation pour enfants "Petits chevaliers" (+ 1€)'}, {'Name': "Abbaye d'Orval", 'Type': 'Local products', 'City': 'Villers-devant-Orval', 'Price': 'adultes\xa07€ • 60+, étudiants 5€ • enfants (7-14) 3€'}, {'Name': 'Abbaye de Maredsous et Le Saint-Joseph', 'Type': 'Local products', 'City': 'Denée', 'Price': 'Entrée libre : Eglise abbatiale

Loop in order to have the different informations for activity related to the remembrance tourism.

In [29]:
path = "C:Users/flori/Desktop/Master 1/Q2/Data Management"
driver = webdriver.Chrome(executable_path=os.path.join(path, 'chromedriver'))
attractions_infos12 = []
url = "https://www.365.be/fr/recherche/?category=1062&range=150&subCategory=1104"
driver.get(url)
time.sleep(5)

soup = BeautifulSoup(driver.page_source, 'html.parser')

events = soup.find('div', class_='main-content__body').find('div', class_='row cards').find_all('article', class_='col-12 col-md-6 col-lg-6 col-xl-4')

for event in events:
    full_event_url = f"https://www.365.be{event.find('a')['href']}"
    driver.get(full_event_url)
    time.sleep(2)
    soup = BeautifulSoup(driver.page_source, 'html.parser')
    
    try:
        attraction_name = soup.find('h1', class_="attraction-name").text.strip()
    except:
        attraction_name = "Not available"
    try:
        attraction_type = "Remembrance tourism"
    except:
        attraction_type = "Not available"
    try:
        city_name = soup.find('span', {'itemprop': 'addressLocality'}).text.strip()
    except:
        city_name = "Not available"
    try:
        price_label = soup.find("span", class_="label", text="Prix").find_next_sibling("div").text.strip()
    except:
        price_label = "Not available"
    attractions_infos12.append({'Name' : attraction_name,'Type': attraction_type, 'City' : city_name, 'Price' : price_label})
print(attractions_infos12)
driver.quit()

C:\Users\flori\AppData\Local\Temp\ipykernel_16500\2800191380.py:2: DeprecationWarning: executable_path has been deprecated, please pass in a Service object
  driver = webdriver.Chrome(executable_path=os.path.join(path, 'chromedriver'))
Incompatible release of chromedriver (version 110.0.5481.77) detected in PATH: C:\Users\flori\Desktop\Master 1\Q2\Data Management\chromedriver.exe


[{'Name': 'Domaine de la bataille de Waterloo 1815', 'Type': 'Remembrance tourism', 'City': 'Braine-l’Alleud', 'Price': 'Ticket Mémorial : adultes à partir de 19,50€ • 60+ à partir de 17,50€ • enfants (10-17) à partir de 9,75€ • enfants (<10 ans) : gratuit • Pass 1815 - 3 musées : www.waterloo1815.be'}, {'Name': 'Musée Wellington', 'Type': 'Remembrance tourism', 'City': 'Waterloo', 'Price': 'adultes 10€ • 60+, étudiants, enfants (10-17) : 8€ • enfants (<10) gratuit   Pass 1815: www.museewellington.be'}, {'Name': 'Plugstreet 14-18 experience', 'Type': 'Remembrance tourism', 'City': 'Ploegsteert', 'Price': 'adultes 7€ • enfants (<12), étudiants, PMR, 60+ : 5€ • enfants (<6) gratuit'}, {'Name': 'Bastogne War Museum', 'Type': 'Remembrance tourism', 'City': 'Bastogne', 'Price': 'adultes 22€ • 65+ 19€ • étudiants 14€ • PMR 14€ • enfants (6-18) 12,50€ • enfants (< 6) gratuit'}, {'Name': 'Citadelle de Dinant', 'Type': 'Remembrance tourism', 'City': 'Dinant', 'Price': "Avec l'ascension en télép

Loop in order to have the different informations for activity related to the UNESCO.

In [31]:
path = "C:Users/flori/Desktop/Master 1/Q2/Data Management"
driver = webdriver.Chrome(executable_path=os.path.join(path, 'chromedriver'))
attractions_infos13 = []
url = "https://www.365.be/fr/recherche/?category=1062&range=150&subCategory=1097"
driver.get(url)
time.sleep(5)

soup = BeautifulSoup(driver.page_source, 'html.parser')

events = soup.find('div', class_='main-content__body').find('div', class_='row cards').find_all('article', class_='col-12 col-md-6 col-lg-6 col-xl-4')

for event in events:
    full_event_url = f"https://www.365.be{event.find('a')['href']}"
    driver.get(full_event_url)
    time.sleep(2)
    soup = BeautifulSoup(driver.page_source, 'html.parser')
    
    try:
        attraction_name = soup.find('h1', class_="attraction-name").text.strip()
    except:
        attraction_name = "Not available"
    try:
        attraction_type = "UNESCO"
    except:
        attraction_type = "Not available"
    try:
        city_name = soup.find('span', {'itemprop': 'addressLocality'}).text.strip()
    except:
        city_name = "Not available"
    try:
        price_label = soup.find("span", class_="label", text="Prix").find_next_sibling("div").text.strip()
    except:
        price_label = "Not available"
    attractions_infos13.append({'Name' : attraction_name,'Type': attraction_type, 'City' : city_name, 'Price' : price_label})
print(attractions_infos13)
driver.quit()

C:\Users\flori\AppData\Local\Temp\ipykernel_16500\3708585189.py:2: DeprecationWarning: executable_path has been deprecated, please pass in a Service object
  driver = webdriver.Chrome(executable_path=os.path.join(path, 'chromedriver'))
Incompatible release of chromedriver (version 110.0.5481.77) detected in PATH: C:\Users\flori\Desktop\Master 1\Q2\Data Management\chromedriver.exe


[{'Name': 'Le Bois du Cazier', 'Type': 'UNESCO', 'City': 'Marcinelle', 'Price': 'adultes 9€\xa0•\xa060+ 8€ • Jeunes et étudiants : 5€\xa0•\xa0enfants (<6) : gratuit   Audioguide : 1€ • Livret-jeu : 1€ • Appli gratuite'}, {'Name': 'Mundaneum', 'Type': 'UNESCO', 'City': 'Mons', 'Price': "adultes 9€ • 65+ : 7€ • demandeurs d'emploi, étudiants, enfants (>6) : 5€ • enfants (<6) gratuit   Gratuit le 1er dimanche du mois"}, {'Name': 'Musée de la Mine et du Développement Durable - Bois-du-Luc', 'Type': 'UNESCO', 'City': 'Houdeng-Aimeries', 'Price': "Visite libre : adultes 9€ / 60+ 8€ / étudiants et enfants (<18) : 6€ / enfants (<6) : gratuit / PMR 4€   Visite avec audioguide : + 1€ p.p. • Forfait famille : 50% sur le prix d'entrée à partir du 2ème enfant • Article 27 : visite libre, guidée ou audioguidée à 1,25€"}, {'Name': 'Blegny-Mine', 'Type': 'UNESCO', 'City': 'Blegny', 'Price': "La visite guidée de la mine (visite libre de l'exposition permanente et biotope du terril inclus) : adultes 15€

Loop in order to have the different informations for activity related to the caves.

In [33]:
path = "C:Users/flori/Desktop/Master 1/Q2/Data Management"
driver = webdriver.Chrome(executable_path=os.path.join(path, 'chromedriver'))
attractions_infos14 = []
url = "https://www.365.be/fr/recherche/?category=1063&range=150&subCategory=1071"
driver.get(url)
time.sleep(5)

soup = BeautifulSoup(driver.page_source, 'html.parser')

events = soup.find('div', class_='main-content__body').find('div', class_='row cards').find_all('article', class_='col-12 col-md-6 col-lg-6 col-xl-4')

for event in events:
    full_event_url = f"https://www.365.be{event.find('a')['href']}"
    driver.get(full_event_url)
    time.sleep(2)
    soup = BeautifulSoup(driver.page_source, 'html.parser')
    
    try:
        attraction_name = soup.find('h1', class_="attraction-name").text.strip()
    except:
        attraction_name = "Not available"
    try:
        attraction_type = "Caves"
    except:
        attraction_type = "Not available"
    try:
        city_name = soup.find('span', {'itemprop': 'addressLocality'}).text.strip()
    except:
        city_name = "Not available"
    try:
        price_label = soup.find("span", class_="label", text="Prix").find_next_sibling("div").text.strip()
    except:
        price_label = "Not available"
    attractions_infos14.append({'Name' : attraction_name,'Type': attraction_type, 'City' : city_name, 'Price' : price_label})
print(attractions_infos14)
driver.quit()

C:\Users\flori\AppData\Local\Temp\ipykernel_16500\1523588794.py:2: DeprecationWarning: executable_path has been deprecated, please pass in a Service object
  driver = webdriver.Chrome(executable_path=os.path.join(path, 'chromedriver'))
Incompatible release of chromedriver (version 110.0.5481.77) detected in PATH: C:\Users\flori\Desktop\Master 1\Q2\Data Management\chromedriver.exe


[{'Name': 'Domaine des Grottes de Han - La Grotte de Han', 'Type': 'Caves', 'City': 'Han-sur-Lesse', 'Price': 'adultes 27€ - enfants (4-11 ans) 21€ - enfants (<4 ans) gratuit   PASSHAN : adultes 39,50€ - enfants (4-11 ans) 29,50€ - enfants (<4 ans) gratuit'}, {'Name': 'Grotte de Lorette-Rochefort', 'Type': 'Caves', 'City': 'Rochefort', 'Price': 'adultes 12,50€ • enfants (4-11 ans) 8€ • enfants (< 4 ans) gratuit'}, {'Name': 'Grotte de Comblain', 'Type': 'Caves', 'City': 'Comblain-au-Pont', 'Price': 'adultes (12 ans et >) 9,50€ • enfants (4-11 ans) et 65 + : 7,50€ • enfants (< 4 ans) gratuit'}, {'Name': 'Grottes de Hotton', 'Type': 'Caves', 'City': 'Hotton', 'Price': 'adultes 12€ • enfants (3-11)\xa09€ • Combiné avec RIVEO, Parc Chlorophylle et Parc des Topiaires : réduction sur le 2e et le 3e site visité'}, {'Name': "Grottes de Neptune dites Grottes de l'Adujoir", 'Type': 'Caves', 'City': 'Petigny', 'Price': "adultes 9€ - enfants (4 à 12ans) 5,50€ • Combiné avec Brûly-de-Pesche 1940 (un

Loop in order to have the different informations for activity related to the gardens, parks and natural sites.

In [35]:
path = "C:Users/flori/Desktop/Master 1/Q2/Data Management"
driver = webdriver.Chrome(executable_path=os.path.join(path, 'chromedriver'))
attractions_infos15 = []
url = "https://www.365.be/fr/recherche/?category=1063&range=150&subCategory=1105"
driver.get(url)
time.sleep(5)

soup = BeautifulSoup(driver.page_source, 'html.parser')

events = soup.find('div', class_='main-content__body').find('div', class_='row cards').find_all('article', class_='col-12 col-md-6 col-lg-6 col-xl-4')

for event in events:
    full_event_url = f"https://www.365.be{event.find('a')['href']}"
    driver.get(full_event_url)
    time.sleep(2)
    soup = BeautifulSoup(driver.page_source, 'html.parser')
    
    try:
        attraction_name = soup.find('h1', class_="attraction-name").text.strip()
    except:
        attraction_name = "Not available"
    try:
        attraction_type = "Gardens, parks and natural sites"
    except:
        attraction_type = "Not available"
    try:
        city_name = soup.find('span', {'itemprop': 'addressLocality'}).text.strip()
    except:
        city_name = "Not available"
    try:
        price_label = soup.find("span", class_="label", text="Prix").find_next_sibling("div").text.strip()
    except:
        price_label = "Not available"
    attractions_infos15.append({'Name' : attraction_name,'Type': attraction_type, 'City' : city_name, 'Price' : price_label})
print(attractions_infos15)
driver.quit()

C:\Users\flori\AppData\Local\Temp\ipykernel_16500\2440010198.py:2: DeprecationWarning: executable_path has been deprecated, please pass in a Service object
  driver = webdriver.Chrome(executable_path=os.path.join(path, 'chromedriver'))
Incompatible release of chromedriver (version 110.0.5481.77) detected in PATH: C:\Users\flori\Desktop\Master 1\Q2\Data Management\chromedriver.exe


[{'Name': 'Parc des Topiaires', 'Type': 'Gardens, parks and natural sites', 'City': 'Durbuy', 'Price': 'adultes 4,50€ • 60+ 4€ • enfants (6-12) 1€ • enfants (<6) gratuit'}, {'Name': 'Aquascope Etang de Virelles', 'Type': 'Gardens, parks and natural sites', 'City': 'Virelles', 'Price': 'Adultes 9€\xa0-\xa0Etudiants 6€ - Enfants (6-12) 5€ - Enfants (<6) gratuit   Abonnement annuel individuel :\xa0adultes 20€ - enfants 12€'}, {'Name': "Les Jardins d'eau d'Annevoie", 'Type': 'Gardens, parks and natural sites', 'City': 'Annevoie', 'Price': 'adultes 10€ • étudiants 7€ • enfants 6€ • PMR 6€'}, {'Name': 'Parc de Furfooz', 'Type': 'Gardens, parks and natural sites', 'City': 'Furfooz', 'Price': 'adultes 5€ • enfants 1€'}, {'Name': 'Les Jardins du Lac de Bambois', 'Type': 'Gardens, parks and natural sites', 'City': 'Fosses-la-Ville', 'Price': 'adultes 5,50€ • 60+ 4,50€ • enfants (3-11) 4€ • enfants (0-2) gratuit   PMR\xa0adultes 4,50€ • PMR enfants 3€'}, {'Name': 'Maison du Parc - Botrange', 'Typ

Loop in order to have the different informations for activity related to the animal parks, zoos and aquariums.

In [37]:
path = "C:Users/flori/Desktop/Master 1/Q2/Data Management"
driver = webdriver.Chrome(executable_path=os.path.join(path, 'chromedriver'))
attractions_infos16 = []
url = "https://www.365.be/fr/recherche/?category=1063&range=150&subCategory=1106"
driver.get(url)
time.sleep(5)

soup = BeautifulSoup(driver.page_source, 'html.parser')

events = soup.find('div', class_='main-content__body').find('div', class_='row cards').find_all('article', class_='col-12 col-md-6 col-lg-6 col-xl-4')

for event in events:
    full_event_url = f"https://www.365.be{event.find('a')['href']}"
    driver.get(full_event_url)
    time.sleep(2)
    soup = BeautifulSoup(driver.page_source, 'html.parser')
    
    try:
        attraction_name = soup.find('h1', class_="attraction-name").text.strip()
    except:
        attraction_name = "Not available"
    try:
        attraction_type = "Animal parks, zoos and aquariums"
    except:
        attraction_type = "Not available"
    try:
        city_name = soup.find('span', {'itemprop': 'addressLocality'}).text.strip()
    except:
        city_name = "Not available"
    try:
        price_label = soup.find("span", class_="label", text="Prix").find_next_sibling("div").text.strip()
    except:
        price_label = "Not available"
    attractions_infos16.append({'Name' : attraction_name,'Type': attraction_type, 'City' : city_name, 'Price' : price_label})
print(attractions_infos16)
driver.quit()

C:\Users\flori\AppData\Local\Temp\ipykernel_16500\4167703794.py:2: DeprecationWarning: executable_path has been deprecated, please pass in a Service object
  driver = webdriver.Chrome(executable_path=os.path.join(path, 'chromedriver'))
Incompatible release of chromedriver (version 110.0.5481.77) detected in PATH: C:\Users\flori\Desktop\Master 1\Q2\Data Management\chromedriver.exe


[{'Name': 'Domaine des Grottes de Han - Le Parc Animalier', 'Type': 'Animal parks, zoos and aquariums', 'City': 'Han-sur-Lesse', 'Price': 'adultes 27€ - enfants (4-11 ans) 21€ - enfants (<4 ans) gratuit   PASSHAN : adultes 39,50€ - enfants (4-11 ans) 29,50€ - enfants (<4 ans) gratuit'}, {'Name': 'Pairi Daiza', 'Type': 'Animal parks, zoos and aquariums', 'City': 'Brugelette', 'Price': 'Tarifs valables pour les achats en ligne : enfants (<3 ans) : gratuit • enfants (3-11 ans) : 33€   adultes (12-59 ans) : 39€ • seniors (60 ans et +) : 37€'}, {'Name': 'RIVEO', 'Type': 'Animal parks, zoos and aquariums', 'City': 'Hotton', 'Price': 'Aquariums et exposition : adultes\xa08€ - 65+, enfants (3-12) :\xa06€ - Famille de 4 personnes : 6€ pp. • Minigolf + jardins : adultes\xa07€ - 65+, enfants (3-12) : 5€ - Famille de 4 personnes : 5€ pp. • Combiné (aquariums, expo & minigolf): adultes 13€ - 65+, enfants, (3-12) :\xa09€ - Famille de 4 personnes : 9€ pp. • Combiné avec les Grottes de Hotton, le Parc

Loop in order to have the different informations for activity related to the tourist boats and cruises.

In [39]:
path = "C:Users/flori/Desktop/Master 1/Q2/Data Management"
driver = webdriver.Chrome(executable_path=os.path.join(path, 'chromedriver'))
attractions_infos17 = []
url = "https://www.365.be/fr/recherche/?category=1064&range=150&subCategory=1108"
driver.get(url)
time.sleep(5)

soup = BeautifulSoup(driver.page_source, 'html.parser')

events = soup.find('div', class_='main-content__body').find('div', class_='row cards').find_all('article', class_='col-12 col-md-6 col-lg-6 col-xl-4')

for event in events:
    full_event_url = f"https://www.365.be{event.find('a')['href']}"
    driver.get(full_event_url)
    time.sleep(2)
    soup = BeautifulSoup(driver.page_source, 'html.parser')
    
    try:
        attraction_name = soup.find('h1', class_="attraction-name").text.strip()
    except:
        attraction_name = "Not available"
    try:
        attraction_type = "Tourist boats and cruises"
    except:
        attraction_type = "Not available"
    try:
        city_name = soup.find('span', {'itemprop': 'addressLocality'}).text.strip()
    except:
        city_name = "Not available"
    try:
        price_label = soup.find("span", class_="label", text="Prix").find_next_sibling("div").text.strip()
    except:
        price_label = "Not available"
    attractions_infos17.append({'Name' : attraction_name,'Type': attraction_type, 'City' : city_name, 'Price' : price_label})
print(attractions_infos17)
driver.quit()

C:\Users\flori\AppData\Local\Temp\ipykernel_16500\4093329256.py:2: DeprecationWarning: executable_path has been deprecated, please pass in a Service object
  driver = webdriver.Chrome(executable_path=os.path.join(path, 'chromedriver'))
Incompatible release of chromedriver (version 110.0.5481.77) detected in PATH: C:\Users\flori\Desktop\Master 1\Q2\Data Management\chromedriver.exe


[{'Name': 'Crocodile Rouge', 'Type': 'Tourist boats and cruises', 'City': 'Boussu-lez-Walcourt', 'Price': 'Crocodile Rouge : adultes 23€ • enfants (4-12) 21€'}, {'Name': 'Dinant Nautique - Bateaux électriques sans permis', 'Type': 'Tourist boats and cruises', 'City': 'Dinant', 'Price': '59€ par bateau (maximum 7 personnes à bord)'}, {'Name': 'Dinant Evasion - Croisières sur la Meuse', 'Type': 'Tourist boats and cruises', 'City': 'Dinant', 'Price': '1) Dinant-Anseremme : adultes 11€ - 60+ 9,50€ - enfants 8€ • 2) Dinant-Freÿr : adultes 18€ - 60+ 16€ - enfants 14€'}, {'Name': 'Bateaux électriques sur le Canal du Centre historique', 'Type': 'Tourist boats and cruises', 'City': 'Houdeng-Goegnies', 'Price': 'Prix par heure, par bateau : 45€ (5 places) • 60€ (7 places) • 80€ (9 places)'}, {'Name': 'Bateaux Meuse-Ardennes', 'Type': 'Tourist boats and cruises', 'City': 'Namur', 'Price': '1) Namur Sambre et Meuse : adultes 10€ - 60+ 9€ - enfants 7€   2) Namur-Wépion et retour : adultes 15€ - 60+

Loop in order to have the different informations for activity related to the draisines and tourist trains.

In [41]:
path = "C:Users/flori/Desktop/Master 1/Q2/Data Management"
driver = webdriver.Chrome(executable_path=os.path.join(path, 'chromedriver'))
attractions_infos18 = []
url = "https://www.365.be/fr/recherche/?category=1064&range=150&subCategory=9941"
driver.get(url)
time.sleep(5)

soup = BeautifulSoup(driver.page_source, 'html.parser')

events = soup.find('div', class_='main-content__body').find('div', class_='row cards').find_all('article', class_='col-12 col-md-6 col-lg-6 col-xl-4')

for event in events:
    full_event_url = f"https://www.365.be{event.find('a')['href']}"
    driver.get(full_event_url)
    time.sleep(2)
    soup = BeautifulSoup(driver.page_source, 'html.parser')
    
    try:
        attraction_name = soup.find('h1', class_="attraction-name").text.strip()
    except:
        attraction_name = "Not available"
    try:
        attraction_type = "Draisines and tourist trains"
    except:
        attraction_type = "Not available"
    try:
        city_name = soup.find('span', {'itemprop': 'addressLocality'}).text.strip()
    except:
        city_name = "Not available"
    try:
        price_label = soup.find("span", class_="label", text="Prix").find_next_sibling("div").text.strip()
    except:
        price_label = "Not available"
    attractions_infos18.append({'Name' : attraction_name,'Type': attraction_type, 'City' : city_name, 'Price' : price_label})
print(attractions_infos18)
driver.quit()

C:\Users\flori\AppData\Local\Temp\ipykernel_16500\1666456046.py:2: DeprecationWarning: executable_path has been deprecated, please pass in a Service object
  driver = webdriver.Chrome(executable_path=os.path.join(path, 'chromedriver'))
Incompatible release of chromedriver (version 110.0.5481.77) detected in PATH: C:\Users\flori\Desktop\Master 1\Q2\Data Management\chromedriver.exe


[{'Name': 'Chemin de fer à vapeur des 3 vallées', 'Type': 'Draisines and tourist trains', 'City': 'Mariembourg', 'Price': 'Aller-retour : adultes 15€ - enfants (6-12) 10€'}, {'Name': 'Draisines de la Molignée', 'Type': 'Draisines and tourist trains', 'City': 'Onhaye', 'Price': 'Prix par draisine (max. 4 personnes) : 22€'}, {'Name': 'Téléphérique de la Citadelle de Namur', 'Type': 'Draisines and tourist trains', 'City': 'Namur', 'Price': 'adultes : aller 5€ / aller-retour 7,50€ • seniors: aller 4€ / aller-retour 6€   étudiants, jeunes et enfants (6+) : aller 4€ / aller-retour 5,50€ • enfants (4-5 ans) et PMR : aller 3,50€ / aller-retour 5€ • enfants (0-3 ans) : gratuit.'}, {'Name': 'Train Touristique de Rochefort', 'Type': 'Draisines and tourist trains', 'City': 'Rochefort', 'Price': 'adultes 7€ • enfants (3-12) 4€ • 60+, étudiants (<26), PMR : 6€'}, {'Name': 'Railbike des Hautes Fagnes', 'Type': 'Draisines and tourist trains', 'City': 'Leykaul - Elsenborn', 'Price': '40€ par draisine (

Loop in order to have the different informations for activity related to the kayaks.

In [43]:
path = "C:Users/flori/Desktop/Master 1/Q2/Data Management"
driver = webdriver.Chrome(executable_path=os.path.join(path, 'chromedriver'))
attractions_infos19 = []
url = "https://www.365.be/fr/recherche/?category=1064&range=150&subCategory=9940"
driver.get(url)
time.sleep(5)

soup = BeautifulSoup(driver.page_source, 'html.parser')

events = soup.find('div', class_='main-content__body').find('div', class_='row cards').find_all('article', class_='col-12 col-md-6 col-lg-6 col-xl-4')

for event in events:
    full_event_url = f"https://www.365.be{event.find('a')['href']}"
    driver.get(full_event_url)
    time.sleep(2)
    soup = BeautifulSoup(driver.page_source, 'html.parser')
    
    try:
        attraction_name = soup.find('h1', class_="attraction-name").text.strip()
    except:
        attraction_name = "Not available"
    try:
        attraction_type = "Kayaks"
    except:
        attraction_type = "Not available"
    try:
        city_name = soup.find('span', {'itemprop': 'addressLocality'}).text.strip()
    except:
        city_name = "Not available"
    try:
        price_label = soup.find("span", class_="label", text="Prix").find_next_sibling("div").text.strip()
    except:
        price_label = "Not available"
    attractions_infos19.append({'Name' : attraction_name,'Type': attraction_type, 'City' : city_name, 'Price' : price_label})
print(attractions_infos19)
driver.quit()

C:\Users\flori\AppData\Local\Temp\ipykernel_16500\2737939731.py:2: DeprecationWarning: executable_path has been deprecated, please pass in a Service object
  driver = webdriver.Chrome(executable_path=os.path.join(path, 'chromedriver'))
Incompatible release of chromedriver (version 110.0.5481.77) detected in PATH: C:\Users\flori\Desktop\Master 1\Q2\Data Management\chromedriver.exe


[{'Name': 'Dinant Evasion - Lesse kayaks', 'Type': 'Kayaks', 'City': 'Anseremme (Dinant)', 'Price': 'à partir de 25,50€ pp.'}, {'Name': "Domaine de Palogne - Descente de l'Ourthe en kayak", 'Type': 'Kayaks', 'City': 'Vieuxville', 'Price': "Kayak biplace (7,8 km) : 21€ pp., transport et gilet de sauvetage compris   Autre activité toute l'année : location de VTT à 21€ et VTT électriques à 36€ (4 heures, casque compris) • Minigolf : 5€"}, {'Name': 'Récréalle Centre Récréatif', 'Type': 'Kayaks', 'City': 'Alle-sur-Semois', 'Price': 'Descente de la Semois en kayak: à partir de\xa020€ pp.  Randonnée en gyropode "Segway" à partir de 25€ pp.   Autres activités sur demande ou sur www.recrealle.com'}, {'Name': 'Kayak Les Remous', 'Type': 'Kayaks', 'City': 'Hamoir', 'Price': "A partir de 22€ pp. pour la descente de Hamoir à Comblain-au-Pont (navette de retour comprise) • A partir de 25€ pp. pour les descentes au départ de Barvaux. Le prix comprend la navette entre le point de départ et le point d'

Loop in order to have the different informations for activity related to the adventure parks.

In [45]:
path = "C:Users/flori/Desktop/Master 1/Q2/Data Management"
driver = webdriver.Chrome(executable_path=os.path.join(path, 'chromedriver'))
attractions_infos20 = []
url = "https://www.365.be/fr/recherche/?category=1064&range=150&subCategory=1109"
driver.get(url)
time.sleep(5)

soup = BeautifulSoup(driver.page_source, 'html.parser')

events = soup.find('div', class_='main-content__body').find('div', class_='row cards').find_all('article', class_='col-12 col-md-6 col-lg-6 col-xl-4')

for event in events:
    full_event_url = f"https://www.365.be{event.find('a')['href']}"
    driver.get(full_event_url)
    time.sleep(2)
    soup = BeautifulSoup(driver.page_source, 'html.parser')
    
    try:
        attraction_name = soup.find('h1', class_="attraction-name").text.strip()
    except:
        attraction_name = "Not available"
    try:
        attraction_type = "Adventure parks"
    except:
        attraction_type = "Not available"
    try:
        city_name = soup.find('span', {'itemprop': 'addressLocality'}).text.strip()
    except:
        city_name = "Not available"
    try:
        price_label = soup.find("span", class_="label", text="Prix").find_next_sibling("div").text.strip()
    except:
        price_label = "Not available"
    attractions_infos20.append({'Name' : attraction_name,'Type': attraction_type, 'City' : city_name, 'Price' : price_label})
print(attractions_infos20)
driver.quit()

C:\Users\flori\AppData\Local\Temp\ipykernel_16500\430487754.py:2: DeprecationWarning: executable_path has been deprecated, please pass in a Service object
  driver = webdriver.Chrome(executable_path=os.path.join(path, 'chromedriver'))
Incompatible release of chromedriver (version 110.0.5481.77) detected in PATH: C:\Users\flori\Desktop\Master 1\Q2\Data Management\chromedriver.exe


[{'Name': 'Aventure Parc Wavre', 'Type': 'Adventure parks', 'City': 'Wavre', 'Price': 'Accès Parcours : 28€ de 5 à 9 ans • 32€ à partir de 10 ans • 2€/personne pour ceux qui accompagnent'}, {'Name': 'Dinant Evasion - Le parc Dinant Aventure', 'Type': 'Adventure parks', 'City': 'Dinant', 'Price': 'à partir de 27€ pp.'}, {'Name': 'Adventure Valley Durbuy', 'Type': 'Adventure parks', 'City': 'Rome (Durbuy)', 'Price': 'Activités à la carte à partir de 5€ • Pass Parc à partir de 10€ pour plusieurs activités'}, {'Name': "Natura Parc - Parc Aventure des Lacs de l'Eau d'Heure", 'Type': 'Adventure parks', 'City': 'Boussu-lez-Walcourt', 'Price': 'adultes 32€ • Jeunes (9-15 ans) 27€ • enfants (3-8 ans) 16€'}, {'Name': 'Ecopark Adventures', 'Type': 'Adventure parks', 'City': 'Tournai', 'Price': 'AVENTURE : adultes (13 ans et +) 27€ - juniors (7-12 ans) 22€ - enfants (5-6 ans) 17€   LEGENDES : 12€ (5-64 ans) - 9€ pour les enfants (2-4 ans) et 65 ans et +   GAMES : 16€ par personne • PASS Aventure +

Loop in order to have the different informations for activity related to the theme and amusement parks.

In [46]:
path = "C:Users/flori/Desktop/Master 1/Q2/Data Management"
driver = webdriver.Chrome(executable_path=os.path.join(path, 'chromedriver'))
attractions_infos21 = []
url = "https://www.365.be/fr/recherche/?category=1064&range=150&subCategory=1110"
driver.get(url)
time.sleep(5)

soup = BeautifulSoup(driver.page_source, 'html.parser')

events = soup.find('div', class_='main-content__body').find('div', class_='row cards').find_all('article', class_='col-12 col-md-6 col-lg-6 col-xl-4')

for event in events:
    full_event_url = f"https://www.365.be{event.find('a')['href']}"
    driver.get(full_event_url)
    time.sleep(2)
    soup = BeautifulSoup(driver.page_source, 'html.parser')
    
    try:
        attraction_name = soup.find('h1', class_="attraction-name").text.strip()
    except:
        attraction_name = "Not available"
    try:
        attraction_type = "Theme and amusement parks"
    except:
        attraction_type = "Not available"
    try:
        city_name = soup.find('span', {'itemprop': 'addressLocality'}).text.strip()
    except:
        city_name = "Not available"
    try:
        price_label = soup.find("span", class_="label", text="Prix").find_next_sibling("div").text.strip()
    except:
        price_label = "Not available"
    attractions_infos21.append({'Name' : attraction_name,'Type': attraction_type, 'City' : city_name, 'Price' : price_label})
print(attractions_infos21)
driver.quit()

C:\Users\flori\AppData\Local\Temp\ipykernel_16500\3155462566.py:2: DeprecationWarning: executable_path has been deprecated, please pass in a Service object
  driver = webdriver.Chrome(executable_path=os.path.join(path, 'chromedriver'))
Incompatible release of chromedriver (version 110.0.5481.77) detected in PATH: C:\Users\flori\Desktop\Master 1\Q2\Data Management\chromedriver.exe


[{'Name': 'Walibi', 'Type': 'Theme and amusement parks', 'City': 'Wavre', 'Price': 'adultes et enfants (>1m40) 45€ • enfants (entre 1m et 1m40) 40€   enfants (<1m) gratuit • seniors (à partir de 55 ans) 37€ • PMR 40€   Vente de tickets en ligne: www.walibi.be'}, {'Name': 'Parc Chlorophylle', 'Type': 'Theme and amusement parks', 'City': 'Dochamps', 'Price': 'adultes 8,50€ • enfants (3-12 ans) 6,50€'}, {'Name': 'Plopsa Coo', 'Type': 'Theme and amusement parks', 'City': 'Coo (Stavelot)', 'Price': 'enfants (<85 cm) gratuit • enfants (85 cm à 1 mètre) et seniors (70+) 14,50€   enfants (>1 mètre) et adultes 29,99€'}, {'Name': 'Crocodile Rouge', 'Type': 'Theme and amusement parks', 'City': 'Boussu-lez-Walcourt', 'Price': 'Crocodile Rouge : adultes 23€ • enfants (4-12) 21€'}, {'Name': 'Houtopia, univers de sens', 'Type': 'Theme and amusement parks', 'City': 'Houffalize', 'Price': 'adultes et enfants 12€ • 65+ et PMR 10,50€   Familles nombreuses à partir de 3 enfants (3-12 ans) : 10,50€ pp.'}, 

Last Loop in order to have the different informations for activity related to the recreational, nautical and water parks.

In [48]:
path = "C:Users/flori/Desktop/Master 1/Q2/Data Management"
driver = webdriver.Chrome(executable_path=os.path.join(path, 'chromedriver'))
attractions_infos22 = []
url = "https://www.365.be/fr/recherche/?category=1064&range=150&subCategory=1107"
driver.get(url)
time.sleep(5)

soup = BeautifulSoup(driver.page_source, 'html.parser')

events = soup.find('div', class_='main-content__body').find('div', class_='row cards').find_all('article', class_='col-12 col-md-6 col-lg-6 col-xl-4')

for event in events:
    full_event_url = f"https://www.365.be{event.find('a')['href']}"
    driver.get(full_event_url)
    time.sleep(2)
    soup = BeautifulSoup(driver.page_source, 'html.parser')
    
    try:
        attraction_name = soup.find('h1', class_="attraction-name").text.strip()
    except:
        attraction_name = "Not available"
    try:
        attraction_type = "Recreational, nautical and water parks"
    except:
        attraction_type = "Not available"
    try:
        city_name = soup.find('span', {'itemprop': 'addressLocality'}).text.strip()
    except:
        city_name = "Not available"
    try:
        price_label = soup.find("span", class_="label", text="Prix").find_next_sibling("div").text.strip()
    except:
        price_label = "Not available"
    attractions_infos22.append({'Name' : attraction_name,'Type': attraction_type, 'City' : city_name, 'Price' : price_label})
print(attractions_infos22)
driver.quit()

C:\Users\flori\AppData\Local\Temp\ipykernel_16500\1862427421.py:2: DeprecationWarning: executable_path has been deprecated, please pass in a Service object
  driver = webdriver.Chrome(executable_path=os.path.join(path, 'chromedriver'))
Incompatible release of chromedriver (version 110.0.5481.77) detected in PATH: C:\Users\flori\Desktop\Master 1\Q2\Data Management\chromedriver.exe


[{'Name': 'Domaine de Wégimont', 'Type': 'Recreational, nautical and water parks', 'City': 'Soumagne', 'Price': 'Entrée au Parc : adultes 3€\xa0- enfants (<12) 2€ -\xa0enfants (<3) gratuit   Piscines et Parc : adultes 5€ -\xa065+, enfants (<12) : 3,50€ -\xa0enfants (<3) gratuit   Canotage : 2€ • Mini-golf : 2€ • Natura Foot golf : 8€ le parcours pour 6 personnes   Pétanque : 5€ les 6 boules pour la 1/2 journée • Aire de restauration : 10€ pour 12 personnes'}, {'Name': 'Plopsaqua Hannut-Landen', 'Type': 'Recreational, nautical and water parks', 'City': 'Hannut', 'Price': 'enfants (<85 cm) gratuit • enfants (85 cm à 1 mètre) et seniors (70+) 13,50€   enfants (>1 mètre) et adultes 27,50€'}, {'Name': 'Aqualibi', 'Type': 'Recreational, nautical and water parks', 'City': 'Wavre', 'Price': 'adultes et enfants (>1m20): 25€ • enfants (entre 1m et 1m20): 9,50€   enfants (<1m) gratuit • seniors (à partir de 55 ans):\xa021€ • PMR 21€'}, {'Name': 'The Spin Leisure Park', 'Type': 'Recreational, naut

Let's now combine the different lists according to the activity into a same dataframe. We will sort the dataframe based on the different cities.

In [60]:
df1 = pd.DataFrame(attractions_infos1)
df2 = pd.DataFrame(attractions_infos2)
df3 = pd.DataFrame(attractions_infos3)
df4 = pd.DataFrame(attractions_infos4)
df5 = pd.DataFrame(attractions_infos5)
df6 = pd.DataFrame(attractions_infos6)
df7 = pd.DataFrame(attractions_infos7)
df8 = pd.DataFrame(attractions_infos8)
df9 = pd.DataFrame(attractions_infos9)
df10 = pd.DataFrame(attractions_infos10)
df11 = pd.DataFrame(attractions_infos11)
df12 = pd.DataFrame(attractions_infos12)
df13 = pd.DataFrame(attractions_infos13)
df14 = pd.DataFrame(attractions_infos14)
df15 = pd.DataFrame(attractions_infos15)
df16 = pd.DataFrame(attractions_infos16)
df17 = pd.DataFrame(attractions_infos17)
df18 = pd.DataFrame(attractions_infos18)
df19 = pd.DataFrame(attractions_infos19)
df20 = pd.DataFrame(attractions_infos20)
df21 = pd.DataFrame(attractions_infos21)
df22 = pd.DataFrame(attractions_infos22)
df = pd.concat([df1, df2, df3, df4, df5, df6, df7, df8, df9, df10, df11, df12, df13, df14, df15, df16, df17, df18, df19, df20, df21, df22], ignore_index=True)
df = df.sort_values(by=['City'])
df

Name  \
253                         Récréalle Centre Récréatif   
96                                    Château de Jehay   
138                 Espace muséal d'Andenne - Le Phare   
18                  Espace muséal d'Andenne - Le Phare   
225                       Les Jardins d'eau d'Annevoie   
251                      Dinant Evasion - Lesse kayaks   
92   Musée de la Bière et du Peket. Château de l'Av...   
183  Musée de la Bière et du Peket. Château de l'Av...   
34                                        Musée Gaspar   
19                                 Musée Archéologique   
113                                  Maison des Géants   
10                                 Espace gallo-romain   
14             Archéosite et Musée d'Aubechies-Beloeil   
73             Archéosite et Musée d'Aubechies-Beloeil   
238                          Monde Sauvage Safari Parc   
196                                Bastogne War Museum   
120              Piconrue - Musée de la Grande Ardenne   
202         Bastogne Barracks (War Heritage Institute)   
237                                         Animalaine   
148                                    Tour Salamandre   
208                                    Tour Salamandre   
97                                  Château de Beloeil   
118       Musée international du Carnaval et du Masque   
192                                   Château de Bioul   
107                                   Château de Bioul   
213                                        Blegny-Mine   
174                                        Blegny-Mine   
229                                L’Escale forestière   
90                            Château fort de Bouillon   
236                         Parc animalier de Bouillon   
135                           Château fort de Bouillon   
139                  Archéoscope Godefroid de Bouillon   
182                                  Chimay Experience   
261  Natura Parc - Parc Aventure des Lacs de l'Eau ...   
267                                    Crocodile Rouge   
240                                    Crocodile Rouge   
285                              Lacs de l'Eau d'Heure   
167                                 Barrage et Skywalk   
278                              The Spin Leisure Park   
15                 Maison du patrimoine médiéval mosan   
112                Maison du patrimoine médiéval mosan   
193            Domaine de la bataille de Waterloo 1815   
231                                        Pairi Daiza   
175                                            Atomium   
79                           Belgian Chocolate Village   
80                                Choco Story Brussels   
159                           Planétarium de Bruxelles   
164                                          Autoworld   
206  Musée Royal de l'Armée et d'Histoire militaire...   
152                 Musée belge de la Franc-Maçonnerie   
61                  Musée belge de la Franc-Maçonnerie   
151                    Maison de l'histoire européenne   
149  Parlamentarium : le centre des visiteurs-ses d...   
147                                Porte de Hal (MRAH)   
146                                            Atomium   
145                                       Musée BELvue   
143                        Musée Art & Histoire (MRAH)   
142                               Palais du Coudenberg   
157                     Muséum des Sciences naturelles   
60                                                 KBR   
144                                experience brussels   
46                              Design Museum Brussels   
273                                        Mini-Europe   
22                                Palais du Coudenberg   
271                        Koezio Live Adventure Games   
49                          Musée de la Bande Dessinée   
43            Musées royaux des Beaux-Arts de Belgique   
53             Musée des instruments de musique (MRAH)   
57                         Musée Art & Histoire 

In [61]:
len(df)

289

We can see that we have for now 289 different elements in our dataframe. We will now make some adjustements in order to have a clear dataframe.

In the following code, we are going to extract from the price column the price for adults and children. We will also detail for each price what the code must take into account in order to have the best representation of the prices for the two groups.

In [62]:
df["Price"] = df["Price"].astype(str)

# Define the function to extract price from the label
def extract_price(category, price_label):
    if category == "adultes":
        pattern = f"(adulte|adultes|adult).+?(\d+(?:[\.,]\d{{1,2}})?)\s*(?:€|\(.*?€\))"
    elif category == "enfants":
        pattern = f"(enfant|enfants).+?(\d+(?:[\.,]\d{{1,2}})?)\s*(?:€|\(.*?€\))"
    else:
        pattern = None
        
    if pattern:
        match = re.search(pattern, price_label, flags=re.IGNORECASE)
        if match:
            return match.group(2).replace(',', '.')
    
    if "gratuit" in price_label.lower() or "entrée gratuite" in price_label.lower() or "entrée libre" in price_label.lower():
        return 0
    pp_pattern = r"\b(\d+(?:[,.]\d{1,2})?)\s*€?\s*pp\b"
    pp_match = re.search(pp_pattern, price_label, flags=re.IGNORECASE)
    if pp_match:
        return pp_match.group(1).replace(',', '.')
    if len(price_label.split()) == 1:
        return price_label.replace(',','.')    
    
    return None

# Extract prices for different categories
df["Adults"] = df["Price"].apply(lambda x: extract_price("adultes", x))
df["Children"] = df["Price"].apply(lambda x: extract_price("enfants", x))

We can then display the dataframe and we observe that the two column for the price of adukts and children have been added to the dataframe.

In [63]:
df

Name  \
253                         Récréalle Centre Récréatif   
96                                    Château de Jehay   
138                 Espace muséal d'Andenne - Le Phare   
18                  Espace muséal d'Andenne - Le Phare   
225                       Les Jardins d'eau d'Annevoie   
251                      Dinant Evasion - Lesse kayaks   
92   Musée de la Bière et du Peket. Château de l'Av...   
183  Musée de la Bière et du Peket. Château de l'Av...   
34                                        Musée Gaspar   
19                                 Musée Archéologique   
113                                  Maison des Géants   
10                                 Espace gallo-romain   
14             Archéosite et Musée d'Aubechies-Beloeil   
73             Archéosite et Musée d'Aubechies-Beloeil   
238                          Monde Sauvage Safari Parc   
196                                Bastogne War Museum   
120              Piconrue - Musée de la Grande Ardenne   
202         Bastogne Barracks (War Heritage Institute)   
237                                         Animalaine   
148                                    Tour Salamandre   
208                                    Tour Salamandre   
97                                  Château de Beloeil   
118       Musée international du Carnaval et du Masque   
192                                   Château de Bioul   
107                                   Château de Bioul   
213                                        Blegny-Mine   
174                                        Blegny-Mine   
229                                L’Escale forestière   
90                            Château fort de Bouillon   
236                         Parc animalier de Bouillon   
135                           Château fort de Bouillon   
139                  Archéoscope Godefroid de Bouillon   
182                                  Chimay Experience   
261  Natura Parc - Parc Aventure des Lacs de l'Eau ...   
267                                    Crocodile Rouge   
240                                    Crocodile Rouge   
285                              Lacs de l'Eau d'Heure   
167                                 Barrage et Skywalk   
278                              The Spin Leisure Park   
15                 Maison du patrimoine médiéval mosan   
112                Maison du patrimoine médiéval mosan   
193            Domaine de la bataille de Waterloo 1815   
231                                        Pairi Daiza   
175                                            Atomium   
79                           Belgian Chocolate Village   
80                                Choco Story Brussels   
159                           Planétarium de Bruxelles   
164                                          Autoworld   
206  Musée Royal de l'Armée et d'Histoire militaire...   
152                 Musée belge de la Franc-Maçonnerie   
61                  Musée belge de la Franc-Maçonnerie   
151                    Maison de l'histoire européenne   
149  Parlamentarium : le centre des visiteurs-ses d...   
147                                Porte de Hal (MRAH)   
146                                            Atomium   
145                                       Musée BELvue   
143                        Musée Art & Histoire (MRAH)   
142                               Palais du Coudenberg   
157                     Muséum des Sciences naturelles   
60                                                 KBR   
144                                experience brussels   
46                              Design Museum Brussels   
273                                        Mini-Europe   
22                                Palais du Coudenberg   
271                        Koezio Live Adventure Games   
49                          Musée de la Bande Dessinée   
43            Musées royaux des Beaux-Arts de Belgique   
53             Musée des instruments de musique (MRAH)   
57                         Musée Art & Histoire 

We will then drop the column price. Indeed, this column is no longer useful for the code. The following code also drops any rows that have missing values in either the 'Adults' or 'Children' column.

In [65]:
df.drop('Price', axis=1, inplace=True)
df = df.dropna(subset=["Adults", "Children"])
df

Name  \
253                         Récréalle Centre Récréatif   
96                                    Château de Jehay   
138                 Espace muséal d'Andenne - Le Phare   
18                  Espace muséal d'Andenne - Le Phare   
225                       Les Jardins d'eau d'Annevoie   
251                      Dinant Evasion - Lesse kayaks   
92   Musée de la Bière et du Peket. Château de l'Av...   
183  Musée de la Bière et du Peket. Château de l'Av...   
34                                        Musée Gaspar   
19                                 Musée Archéologique   
113                                  Maison des Géants   
10                                 Espace gallo-romain   
14             Archéosite et Musée d'Aubechies-Beloeil   
73             Archéosite et Musée d'Aubechies-Beloeil   
238                          Monde Sauvage Safari Parc   
196                                Bastogne War Museum   
120              Piconrue - Musée de la Grande Ardenne   
202         Bastogne Barracks (War Heritage Institute)   
237                                         Animalaine   
148                                    Tour Salamandre   
208                                    Tour Salamandre   
97                                  Château de Beloeil   
118       Musée international du Carnaval et du Masque   
192                                   Château de Bioul   
107                                   Château de Bioul   
213                                        Blegny-Mine   
174                                        Blegny-Mine   
229                                L’Escale forestière   
90                            Château fort de Bouillon   
236                         Parc animalier de Bouillon   
135                           Château fort de Bouillon   
139                  Archéoscope Godefroid de Bouillon   
182                                  Chimay Experience   
261  Natura Parc - Parc Aventure des Lacs de l'Eau ...   
267                                    Crocodile Rouge   
240                                    Crocodile Rouge   
285                              Lacs de l'Eau d'Heure   
167                                 Barrage et Skywalk   
278                              The Spin Leisure Park   
15                 Maison du patrimoine médiéval mosan   
112                Maison du patrimoine médiéval mosan   
193            Domaine de la bataille de Waterloo 1815   
231                                        Pairi Daiza   
175                                            Atomium   
79                           Belgian Chocolate Village   
80                                Choco Story Brussels   
159                           Planétarium de Bruxelles   
164                                          Autoworld   
206  Musée Royal de l'Armée et d'Histoire militaire...   
152                 Musée belge de la Franc-Maçonnerie   
61                  Musée belge de la Franc-Maçonnerie   
151                    Maison de l'histoire européenne   
149  Parlamentarium : le centre des visiteurs-ses d...   
147                                Porte de Hal (MRAH)   
146                                            Atomium   
145                                       Musée BELvue   
143                        Musée Art & Histoire (MRAH)   
142                               Palais du Coudenberg   
157                     Muséum des Sciences naturelles   
60                                                 KBR   
144                                experience brussels   
46                              Design Museum Brussels   
273                                        Mini-Europe   
22                                Palais du Coudenberg   
271                        Koezio Live Adventure Games   
49                          Musée de la Bande Dessinée   
43            Musées royaux des Beaux-Arts de Belgique   
53             Musée des instruments de musique (MRAH)   
57                         Musée Art & Histoire 

The following code will remove the euro symbol that could be present with the values in the columns 'Adults' and 'Children', it also converts theses values to integrers and then update the dataframe.

In [55]:
for i in range(len(df['Adults'])):
    if '€' in str(df['Adults'].iloc[i]):
        df['Adults'].iloc[i] = int(str(df['Adults'].iloc[i]).replace('€', ''))
        
for i in range(len(df['Children'])):
    if '€' in str(df['Children'].iloc[i]):
        df['Children'].iloc[i] = int(str(df['Children'].iloc[i]).replace('€', ''))

C:\Users\flori\AppData\Local\Temp\ipykernel_16500\1526985394.py:3: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  df['Adults'].iloc[i] = int(str(df['Adults'].iloc[i]).replace('€', ''))
C:\Users\flori\AppData\Local\Temp\ipykernel_16500\1526985394.py:7: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  df['Children'].iloc[i] = int(str(df['Children'].iloc[i]).replace('€', ''))


In [56]:
df

Name  \
253                         Récréalle Centre Récréatif   
96                                    Château de Jehay   
138                 Espace muséal d'Andenne - Le Phare   
18                  Espace muséal d'Andenne - Le Phare   
225                       Les Jardins d'eau d'Annevoie   
251                      Dinant Evasion - Lesse kayaks   
92   Musée de la Bière et du Peket. Château de l'Av...   
183  Musée de la Bière et du Peket. Château de l'Av...   
34                                        Musée Gaspar   
19                                 Musée Archéologique   
113                                  Maison des Géants   
10                                 Espace gallo-romain   
14             Archéosite et Musée d'Aubechies-Beloeil   
73             Archéosite et Musée d'Aubechies-Beloeil   
238                          Monde Sauvage Safari Parc   
196                                Bastogne War Museum   
120              Piconrue - Musée de la Grande Ardenne   
202         Bastogne Barracks (War Heritage Institute)   
237                                         Animalaine   
148                                    Tour Salamandre   
208                                    Tour Salamandre   
97                                  Château de Beloeil   
118       Musée international du Carnaval et du Masque   
192                                   Château de Bioul   
107                                   Château de Bioul   
213                                        Blegny-Mine   
174                                        Blegny-Mine   
229                                L’Escale forestière   
90                            Château fort de Bouillon   
236                         Parc animalier de Bouillon   
135                           Château fort de Bouillon   
139                  Archéoscope Godefroid de Bouillon   
182                                  Chimay Experience   
261  Natura Parc - Parc Aventure des Lacs de l'Eau ...   
267                                    Crocodile Rouge   
240                                    Crocodile Rouge   
285                              Lacs de l'Eau d'Heure   
167                                 Barrage et Skywalk   
278                              The Spin Leisure Park   
15                 Maison du patrimoine médiéval mosan   
112                Maison du patrimoine médiéval mosan   
193            Domaine de la bataille de Waterloo 1815   
231                                        Pairi Daiza   
175                                            Atomium   
79                           Belgian Chocolate Village   
80                                Choco Story Brussels   
159                           Planétarium de Bruxelles   
164                                          Autoworld   
206  Musée Royal de l'Armée et d'Histoire militaire...   
152                 Musée belge de la Franc-Maçonnerie   
61                  Musée belge de la Franc-Maçonnerie   
151                    Maison de l'histoire européenne   
149  Parlamentarium : le centre des visiteurs-ses d...   
147                                Porte de Hal (MRAH)   
146                                            Atomium   
145                                       Musée BELvue   
143                        Musée Art & Histoire (MRAH)   
142                               Palais du Coudenberg   
157                     Muséum des Sciences naturelles   
60                                                 KBR   
144                                experience brussels   
46                              Design Museum Brussels   
273                                        Mini-Europe   
22                                Palais du Coudenberg   
271                        Koezio Live Adventure Games   
49                          Musée de la Bande Dessinée   
43            Musées royaux des Beaux-Arts de Belgique   
53             Musée des instruments de musique (MRAH)   
57                         Musée Art & Histoire 

Then, we will try to collect the Latitude and longitude of the different cities. Then we will add the column for the Latitude and Longitude in the dataframe. This could be useful in the following parts of the project.

In [13]:
pip install geopy

Note: you may need to restart the kernel to use updated packages.


In [57]:
from geopy.geocoders import Nominatim
geolocator = Nominatim(user_agent='my_app')
def get_coordinates(city):
    try:
        location = geolocator.geocode(city)
        return location.latitude, location.longitude
    except:
        return None, None
df[['Latitude', 'Longitude']] = df['City'].apply(lambda x: pd.Series(get_coordinates(x)))
df

C:\Users\flori\AppData\Local\Temp\ipykernel_16500\3122450696.py:9: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  df[['Latitude', 'Longitude']] = df['City'].apply(lambda x: pd.Series(get_coordinates(x)))
C:\Users\flori\AppData\Local\Temp\ipykernel_16500\3122450696.py:9: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  df[['Latitude', 'Longitude']] = df['City'].apply(lambda x: pd.Series(get_coordinates(x)))


Name  \
253                         Récréalle Centre Récréatif   
96                                    Château de Jehay   
138                 Espace muséal d'Andenne - Le Phare   
18                  Espace muséal d'Andenne - Le Phare   
225                       Les Jardins d'eau d'Annevoie   
251                      Dinant Evasion - Lesse kayaks   
92   Musée de la Bière et du Peket. Château de l'Av...   
183  Musée de la Bière et du Peket. Château de l'Av...   
34                                        Musée Gaspar   
19                                 Musée Archéologique   
113                                  Maison des Géants   
10                                 Espace gallo-romain   
14             Archéosite et Musée d'Aubechies-Beloeil   
73             Archéosite et Musée d'Aubechies-Beloeil   
238                          Monde Sauvage Safari Parc   
196                                Bastogne War Museum   
120              Piconrue - Musée de la Grande Ardenne   
202         Bastogne Barracks (War Heritage Institute)   
237                                         Animalaine   
148                                    Tour Salamandre   
208                                    Tour Salamandre   
97                                  Château de Beloeil   
118       Musée international du Carnaval et du Masque   
192                                   Château de Bioul   
107                                   Château de Bioul   
213                                        Blegny-Mine   
174                                        Blegny-Mine   
229                                L’Escale forestière   
90                            Château fort de Bouillon   
236                         Parc animalier de Bouillon   
135                           Château fort de Bouillon   
139                  Archéoscope Godefroid de Bouillon   
182                                  Chimay Experience   
261  Natura Parc - Parc Aventure des Lacs de l'Eau ...   
267                                    Crocodile Rouge   
240                                    Crocodile Rouge   
285                              Lacs de l'Eau d'Heure   
167                                 Barrage et Skywalk   
278                              The Spin Leisure Park   
15                 Maison du patrimoine médiéval mosan   
112                Maison du patrimoine médiéval mosan   
193            Domaine de la bataille de Waterloo 1815   
231                                        Pairi Daiza   
175                                            Atomium   
79                           Belgian Chocolate Village   
80                                Choco Story Brussels   
159                           Planétarium de Bruxelles   
164                                          Autoworld   
206  Musée Royal de l'Armée et d'Histoire militaire...   
152                 Musée belge de la Franc-Maçonnerie   
61                  Musée belge de la Franc-Maçonnerie   
151                    Maison de l'histoire européenne   
149  Parlamentarium : le centre des visiteurs-ses d...   
147                                Porte de Hal (MRAH)   
146                                            Atomium   
145                                       Musée BELvue   
143                        Musée Art & Histoire (MRAH)   
142                               Palais du Coudenberg   
157                     Muséum des Sciences naturelles   
60                                                 KBR   
144                                experience brussels   
46                              Design Museum Brussels   
273                                        Mini-Europe   
22                                Palais du Coudenberg   
271                        Koezio Live Adventure Games   
49                          Musée de la Bande Dessinée   
43            Musées royaux des Beaux-Arts de Belgique   
53             Musée des instruments de musique (MRAH)   
57                         Musée Art & Histoire 

Finally, we will translate the dataframe into an excel document. It will be easier for us to continue the project.

In [58]:
df.to_excel('webscraping.xlsx', index=False)